test/files/sample.ts

```diff
@@ -1,7 +1,67 @@
 /*
  * sample.1
+ * SampleInterface protocol is a sample among samples.
  *
  * Auto-generated to TypeScript types by avdl-compiler v1.4.1 (https://github.com/keybase/node-avdl-compiler)
  * Input files:
  * - avdl/sample.avdl
  */
+
+import * as keybase1 from 'github.com/keybase/client/go/protocol/keybase1'
+
+/**
+ * Joe is an alias for an int.
+ */
+export type Joe = number
+
+/**
+ * R is a rad record.
+ */
+export type R = {
+  bar: keybase1.UID
+  baz: keybase1.UID
+  woop?: string
+}
+
+export enum Types {
+  NONE = 0,
+  BOZO = 1,
+  BIPPY = 2,
+  AGGLE = 3,
+  FLAGGLE = 4,
+}
+
+export enum EnumNoString {
+  NOSTRING = 0,
+}
+
+export type Boozle = { typ: Types.BOZO, BOZO: number | null } | { typ: Types.BIPPY, BIPPY: string | null } | { typ: Types.AGGLE, AGGLE: number[] | null } | { typ: Types.FLAGGLE, FLAGGLE: boolean[] | null }
```

Comment:
Why is it `BOZO: number | null` instead of `BOZO: number`?

Reply:
I'm not totally sure. That's how we do it in bin/flow.js: https://github.com/keybase/client/blob/master/protocol/bin/flow.js#L380

Reply:
For the record, talked with Danny and it's a point on the spectrum between well-formed and possible. Going to keep it the same as client.